x-pack/plugin/sql/qa/security/src/test/java/org/elasticsearch/xpack/sql/qa/security/RestSqlSecurityIT.java

```diff
@@ -201,9 +202,13 @@ private static void assertResponse(Map<String, Object> expected, Map<String, Obj
             }
         }
 
-        private static Map<String, Object> toMap(Response response) throws IOException {
+        private static Map<String, Object> toMap(Response response, String mode) throws IOException {
             try (InputStream content = response.getEntity().getContent()) {
-                return XContentHelper.convertToMap(JsonXContent.jsonXContent, content, false);
+                if (mode.equalsIgnoreCase("jdbc")) {
```

Comment:
Again, I don't get why check only the mode and not also the binary transfer?

Reply:
For the tests using this method, I haven't introduced the `binary_response` parameter, which means the default behavior is used (returning binary response).